File: Programmers/[67256] 키패드 누르기/woong-jae/67256.js
function solution(numbers, hand) {
    let answer = '';
    
    let leftHand = 10, rightHand = 12;
    numbers.forEach(num => {
        if(num === 0) num = 11; // 0을 11로 처리
        
        if(num % 3 === 1) { // 1, 4, 7
            answer += "L"
            leftHand = num;
        }
        else if(num % 3 === 0) { // 3, 6, 9
            answer += "R"
            rightHand = num;
        }
        else { // 2, 5, 8, 0
            let leftDistance = Math.abs(num - (leftHand % 3 === 1 ? leftHand + 1 : leftHand)) / 3;
            if(leftHand % 3 === 1) leftDistance++;
            let rightDistance = Math.abs(num - (rightHand % 3 === 0 ? rightHand - 1 : rightHand)) / 3;
            if(rightHand % 3 === 0) rightDistance++;
            if(leftDistance < rightDistance) {
                answer += "L";
                leftHand = num;
            }
            else if(leftDistance > rightDistance) {
                answer += "R"
                rightHand = num;
            }
            else {
                if(hand === "right") {
                    answer += "R";
                    rightHand = num;
                }
                else {
                    answer += "L";
                    leftHand = num;
                }
            }
        }
    });
    
    return answer;
}

File: Programmers/[67256] 키패드 누르기/Johoseong/67256.py
def length(num, hand):
    row = abs(((num - 1) // 3) - ((hand - 1) // 3))
    col = abs(((num - 1) % 3) - ((hand - 1) % 3))
    return row + col

def solution(numbers, hand):
    answer = ''

    r_hand = 12
    l_hand = 10

    for n in numbers:
        if n == 1 or n == 4 or n == 7:
            answer += 'L'
            l_hand = n
        elif n == 3 or n == 6 or n == 9:
            answer += 'R'
            r_hand = n
        else:
            tmp = n
            if tmp == 0:
                tmp = 11
            r_len = length(tmp, r_hand)
            l_len = length(tmp, l_hand)

            if r_len < l_len:
                answer += 'R'
                r_hand = tmp
            elif r_len > l_len:
                answer += 'L'
                l_hand = tmp
            else:
                if hand == 'right':
                    answer += 'R'
                    r_hand = tmp
                else:
                    answer += 'L'
                    l_hand = tmp

    return answer
